tgui: Add new and disambiguate old gas colors

esainane opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This PR adds color mappings for new gases, and disambiguates color mappings for old gases where multiple gases used one color - many used `red`, or `purple`.

Most of these have been around for a while, but were missing color mappings in interfaces.

This mainly affects the HFR gas list interface.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

No more falling back to `color={false}` in Hypertorus.js

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. -->

:cl:
code: The tgui ProgressBar can now accept color specifications in hexademical, rgb/rgba, color-\<name\> class, and base CSS named colors.
fix: Add missing gas color mappings to tgui, and disambiguate old mappings. Exotic (and mundane) gases in the SM monitor and HFR interfaces have never been more colorful!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

esainane committed:
tgui: Add new and disambiguate old gas colors

Most of these have been around for a while, but lacked dedicated color
mappings in interfaces.

This mainly affects the HFR gas list interface.

stylemistake reviewed (changes requested):
These colors most probably break the appearance of progress bars in SM monitor. Only names that are listed in styles/colors.scss are valid.

esainane commented:
![cims-colors](https://user-images.githubusercontent.com/14355175/132936910-0da3dcda-c411-4a06-a155-ba12169d9134.png)

stylemistake commented:
Adding these colors to colors.scss is also not an option, because it's adds a lot of CSS size bloat for no good reason.

LemonInTheDark commented:
@stylemistake you happy with this?

esainane commented:
This is not ready to merge, as it currently breaks colored progress bars (such as in SM and HFR display)

LemonInTheDark commented:
Aight

esainane committed:
tgui: ProgressBar: Permit base CSS colors

This extends the definition color handling of ProgressBar to take any
CSS form, including hexadecimal values, rgb()/rgba() functions,
and CSS classes not overridden by CSS classes defined in constants.js.

ProgressBar.stories.js has been extended to show color overriding
features in Kitchen Sink. Leaving the override string blank will
preserve the current behavior, demonstrating automatic color selection
based on ranges. Ranges can also now use CSS base colors in any of the
above formats.

esainane commented:
ProgressBar should now accept a broad set of color specifications, including range specified ones. Kitchen Sink was extended to make this behavior easier to test and preview.

I'm not familiar with Inferno, so any review and commentary would be appreciated!

esainane commented:
![cims-colors-no-scss](https://user-images.githubusercontent.com/14355175/133893334-4e16424e-dba0-457f-bfae-56d1928bca35.png)

stylemistake commented:
Well implemented.
